src/lib/blkalloc/append_blk_allocator.cpp

```diff
@@ -102,95 +79,111 @@ BlkAllocStatus AppendBlkAllocator::alloc(blk_count_t nblks, const blk_alloc_hint
     }
 
     // Push 1 blk to the vector which has all the requested nblks;
-    out_bid = BlkId{m_last_append_offset, nblks, m_chunk_id};
-
-    auto cur_cp = hs()->cp_mgr().cp_guard();
-    m_last_append_offset += nblks;
-    m_freeable_nblks -= nblks;
-
-    // it is guaranteed that dirty buffer always contains updates of current_cp or next_cp, it will
-    // never get dirty buffer from across updates;
-    set_dirty_offset(cur_cp->id() % MAX_CP_COUNT);
+    out_bid = BlkId{m_last_append_offset.fetch_add(nblks), nblks, m_chunk_id};
 
     COUNTER_INCREMENT(m_metrics, num_alloc, 1);
 
     return BlkAllocStatus::SUCCESS;
 }
 
-BlkAllocStatus AppendBlkAllocator::alloc_on_disk(BlkId const&) { return BlkAllocStatus::SUCCESS; }
-
-BlkAllocStatus AppendBlkAllocator::mark_blk_allocated(BlkId const&) { return BlkAllocStatus::SUCCESS; }
-
-void AppendBlkAllocator::free_on_disk(BlkId const&) {}
+// Reserve on disk will update the commit_offset with the new_offset, if its above the current commit_offset.
+BlkAllocStatus AppendBlkAllocator::reserve_on_disk(BlkId const& blkid) {
+    auto cpg = hs()->cp_mgr().cp_guard();
```

Comment:
we need an release assert commit_offset should not across m_last_append_offset. 
This is for self-defense that caller is messing up with blkallocator, e.g. only called reserver_on_disk, not calling reserve_on_cache

Reply:
Done